core/libdeno/binding.cc

```diff
@@ -162,6 +162,68 @@ void ErrorToJSON(const v8::FunctionCallbackInfo<v8::Value>& args) {
   args.GetReturnValue().Set(v8_str(json_string.c_str()));
 }
 
+class GCObserver {
```

Comment:
I'm embarrassed to say but I totally forgot that V8 has this weak callback functionality.

I don't like introducing new nomenclature like "GCObserver". Maybe call it"PersistentWithWeakCallback" ? 

Have you looked at the TracedGlobal class? It seems like that could be used here as well?
Can you move this to its own file, core/libdeno/

Reply:
Seems requires manual setup of a custom `EmbedderHeapTracer` implementation, and closer to `kInternalFields`.

Before experimenting with this I actually want to set if the observed is actually GC-ed after a proper use of callback instead of being resurrected. (Though I'm having trouble now with native syntaxes in snapshots to use the `%DebugTrackRetainingPath` builtin)